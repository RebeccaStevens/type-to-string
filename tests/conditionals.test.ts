import { describe, it } from "vitest";

import { runTests } from "./helpers";

describe("Conditionals", () => {
  it.each([
    "type Test<G> = G extends number ? number : string;",
    "type Test<G> = G extends number ? readonly number[] : readonly string[];",
    "type Test<G> = G extends number ? { foo: number } : { foo: string };",
    "type Test<G> = Readonly<G extends number ? { foo: number } : { foo: string }>;",
  ])("%s", (code) => {
    runTests(code);
  });
});
