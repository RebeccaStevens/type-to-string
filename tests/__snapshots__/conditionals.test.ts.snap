// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Conditionals > type Test<G> = G extends number ? { foo: number } : { foo: string }; 1`] = `
{
  "typeAlias": "Test<G>",
  "typeReference": null,
  "typeToString": "Test<G>",
}
`;

exports[`Conditionals > type Test<G> = G extends number ? { foo: number } : { foo: string }; 2`] = `
{
  "typeNodeAlias": "Test<G>",
  "typeNodeAsWritten": "G extends number ? { foo: number } : { foo: string }",
  "typeNodeToString": null,
  "typeReferenceNode": null,
}
`;

exports[`Conditionals > type Test<G> = G extends number ? number : string; 1`] = `
{
  "typeAlias": "Test<G>",
  "typeReference": null,
  "typeToString": "Test<G>",
}
`;

exports[`Conditionals > type Test<G> = G extends number ? number : string; 2`] = `
{
  "typeNodeAlias": "Test<G>",
  "typeNodeAsWritten": "G extends number ? number : string",
  "typeNodeToString": null,
  "typeReferenceNode": null,
}
`;

exports[`Conditionals > type Test<G> = G extends number ? readonly number[] : readonly string[]; 1`] = `
{
  "typeAlias": "Test<G>",
  "typeReference": null,
  "typeToString": "Test<G>",
}
`;

exports[`Conditionals > type Test<G> = G extends number ? readonly number[] : readonly string[]; 2`] = `
{
  "typeNodeAlias": "Test<G>",
  "typeNodeAsWritten": "G extends number ? readonly number[] : readonly string[]",
  "typeNodeToString": null,
  "typeReferenceNode": null,
}
`;

exports[`Conditionals > type Test<G> = Readonly<G extends number ? { foo: number } : { foo: string }>; 1`] = `
{
  "typeAlias": "Readonly<T>",
  "typeReference": null,
  "typeToString": "Readonly<G extends number ? { foo: number; } : { foo: string; }>",
}
`;

exports[`Conditionals > type Test<G> = Readonly<G extends number ? { foo: number } : { foo: string }>; 2`] = `
{
  "typeNodeAlias": "Test<G>",
  "typeNodeAsWritten": "Readonly<G extends number ? { foo: number } : { foo: string }>",
  "typeNodeToString": null,
  "typeReferenceNode": "Readonly<T>",
}
`;
